### ISSUE TITLE
Crash in IBGMethodSwizzler viewDidAppear

### ISSUE THREAD

**Issue by lha-ryq** [Instabug/Instabug-iOS#159]
We encountered a crash in our app and it looks like it is related with Instabug. So far it occurred only once.

Instabug version: 7.2.6
Device: iPhone 6s
iOS version: 10.2.1

```
Crash due to signal: SIGSEGV(SEGV_MAPERR) at 00000010

0   Instabug                        -[IBGMethodSwizzler viewDidAppear:] (IBGMethodSwizzler.m:177)
1   topspot                         _TToFC7topspot18BaseViewController13viewDidAppearfSbT__merged + 112
2   UIKit                           -[UIViewController _setViewAppearState:isAnimating:] + 848
3   UIKit                           -[UIViewController _endAppearanceTransition:] + 220
4   UIKit                           -[UIPresentationController transitionDidFinish:] + 844
5   UIKit                           __56-[UIPresentationController runTransitionForCurrentState]_block_invoke_2 + 188
6   UIKit                           -[_UIViewControllerTransitionContext completeTransition:] + 116
7   topspot                         (unknown) (OnboardingAnimatedTransitioning.swift:0)
8   UIKit                           -[UIViewAnimationBlockDelegate _didEndBlockAnimation:finished:context:] + 492
9   UIKit                           -[UIViewAnimationState sendDelegateAnimationDidStop:finished:] + 312
10  UIKit                           -[UIViewAnimationState animationDidStop:finished:] + 160
11  QuartzCore                      CA::Layer::run_animation_callbacks() + 260
12  libdispatch.dylib               _dispatch_client_callout + 16
13  libdispatch.dylib               _dispatch_main_queue_callback_4CF + 1000
14  CoreFoundation                  __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 12
15  CoreFoundation                  __CFRunLoopRun + 1660
16  CoreFoundation                  CFRunLoopRunSpecific + 444
17  GraphicsServices                GSEventRunModal + 180
18  UIKit                           -[UIApplication _run] + 684
19  UIKit                           UIApplicationMain + 208
20  topspot                         main
21  libdyld.dylib                   start + 4
```

**Comment by Kmohamed** [comment]
Hello @lha-ryq
Actually this crash is caused by a bad memory address and it's not necessary to be caused by Instabug because we are swizzling each viewDidAppear.
so If the crash occurred in viewDidAppear for some reason you will see Instabug swizzle stack frame in crash stack trace.
So I would appreciate if you send a steps to reproduce.

**Comment by lha-ryq** [comment]
Hi @Kmohamed
This crash happened to our users only once. It occurred during first application startup and just after performing custom transition.
Our implementation for viewDidAppear of the view controller which is presented after transition is almost empty.
I am going to execute more tests to reproduce this crash.

**Comment by SudoPlz** [comment]
I had a random crash as well, and I received that stack trace, not sure if it's the same problem:

```
EXC_BAD_ACCESS Attempted to dereference garbage pointer 0x10. 
    Frameworks/Instabug.framework/Instabug unzRepair
    Frameworks/UIKit.framework/UIKit <redacted>
    Frameworks/UIKit.framework/UIKit <redacted>
    Frameworks/UIKit.framework/UIKit <redacted>
    Frameworks/CoreFoundation.framework/CoreFoundation <redacted>
    Frameworks/CoreFoundation.framework/CoreFoundation <redacted>
    Frameworks/CoreFoundation.framework/CoreFoundation <redacted>
    Frameworks/CoreFoundation.framework/CoreFoundation CFRunLoopRunSpecific
    PrivateFrameworks/GraphicsServices.framework/GraphicsServices GSEventRunModal
    Frameworks/UIKit.framework/UIKit UIApplicationMain
    TestApp mh_execute_header
    /usr/lib/system/libdyld.dylib <redacted>
```

**Comment by Kmohamed** [comment]
@SudoPlz Could you please send to me full stack trace ?

**Comment by SudoPlz** [comment]
@Kmohamed that's all I have unfortunately :/ Here's another one too:

```
EXC_BAD_ACCESS Attempted to dereference garbage pointer 0xc. 
    Frameworks/Instabug.framework/Instabug unzRepair
    Frameworks/UIKit.framework/UIKit -[UINavigationController viewDidAppear:]
    Frameworks/UIKit.framework/UIKit -[UIViewController _setViewAppearState:isAnimating:]
    Frameworks/UIKit.framework/UIKit -[UIViewController _endAppearanceTransition:]
    TestApp ZN5folly6detail15str_to_integralIxEENS_8ExpectedIT_NS_14ConversionCodeEEEPNS_5RangeIPKcEE
    Frameworks/UIKit.framework/UIKit -[UIViewController _setViewAppearState:isAnimating:]
    Frameworks/UIKit.framework/UIKit ___64-[UIViewController viewDidMoveToWindow:shouldAppearOrDisappear:]_block_invoke
    Frameworks/UIKit.framework/UIKit -[UIViewController _executeAfterAppearanceBlock]
    Frameworks/UIKit.framework/UIKit __runAfterCACommitDeferredBlocks
    Frameworks/UIKit.framework/UIKit __cleanUpAfterCAFlushAndRunDeferredBlocks
    Frameworks/UIKit.framework/UIKit __afterCACommitHandler
    Frameworks/CoreFoundation.framework/CoreFoundation ___CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__
    Frameworks/CoreFoundation.framework/CoreFoundation ___CFRunLoopDoObservers
    Frameworks/CoreFoundation.framework/CoreFoundation ___CFRunLoopRun
    Frameworks/CoreFoundation.framework/CoreFoundation CFRunLoopRunSpecific
    Frameworks/CoreFoundation.framework/CoreFoundation CFRunLoopRunInMode
    PrivateFrameworks/GraphicsServices.framework/GraphicsServices GSEventRunModal
    Frameworks/UIKit.framework/UIKit UIApplicationMain
    TestApp mh_execute_header
```

**Comment by alexsanderkhitev** [comment]
Hello @Kmohamed ! We also got one similar app crash from the user, and once managed to get it in Xcode.

At us it happens during the first loading of application.

**Comment by Kmohamed** [comment]
@alexsanderkhitev
I am sorry for that but could you send to me stack trace ? And are you able to reproduce it ?

**Comment by alexsanderkhitev** [comment]
@Kmohamed 
I got this in the Xcode console - "Instabug was compiled with optimization - stepping may behave oddly; variables may not be available"

 Sorry, but I can not provide this, because I did not save and did only a screenshot, I can send you a screenshot of this app crash.

**Comment by alexsanderkhitev** [comment]
@Kmohamed Thanks for the prompt reply, do you have any suggestions on how to eliminate this in the shortest time (1-2 hours)?

**Comment by Kmohamed** [comment]
I am on it now but please send to me the screenshot

**Comment by alexsanderkhitev** [comment]
@Kmohamed 
![instabug](https://user-images.githubusercontent.com/12511909/28153785-6aebe560-67b8-11e7-8e91-f165d3168769.png)

**Comment by alexsanderkhitev** [comment]
@Kmohamed This happened during the first run, I did not have time to press the button to enter the application and the fall occurred.

**Comment by alexsanderkhitev** [comment]
@Kmohamed The user had this happen at 2 seconds when he was already inside the application, the app crash happens randomly

**Comment by Kmohamed** [comment]
@alexsanderkhitev could you list your third parties ?

**Comment by alexsanderkhitev** [comment]
@Kmohamed yes, I can

 # Firebase
    pod 'Firebase/Core'
    pod 'Firebase/Auth'
    pod 'Firebase/Database'
    pod 'Firebase/Storage'
    # Push notification
    pod 'Firebase/Core'
    pod 'Firebase/Messaging'
    
    # Firebase Additional
    pod 'GeoFire', :git => 'https://github.com/firebase/geofire-objc.git', :commit => '96d6463f59016785f9c5ae9b669019c44786a23f'
    
    # JSON, Models
    pod 'ObjectMapper', '~> 2.2'
    
    # UI
    pod 'ActiveLabel'
    pod 'DynamicColor', '~> 3.2.1'
    pod 'Koloda', '~> 4.0'
    pod 'SVProgressHUD', :git => 'https://github.com/SVProgressHUD/SVProgressHUD.git', :commit => 'f7c8d0d5c9ec60aa7e0682d2f21cb60e8f83d084'
    pod 'RangeSeekSlider'
    pod 'DGRunkeeperSwitch', '~> 1.1.4'

    # Eureka
    pod 'Eureka', '~> 3.0.0'
    pod 'SKPhotoBrowser'
    pod 'MIBlurPopup'
    pod 'XLPagerTabStrip' #, '~> 7.0'

    
    # Social
    # - Facebook
    pod 'FBSDKCoreKit'
    pod 'FBSDKLoginKit'
    pod 'FBSDKShareKit'
    
    # Database
    pod 'RealmSwift'
    
    # Date
    pod 'SwiftDate', '~> 4.0'
    
    # Network
    pod 'Alamofire', '~> 4.4'
    pod 'Kingfisher'
    
    # Api
    pod 'QuadratTouch', :git => 'https://github.com/Constantine-Fry/das-quadrat', :branch => 'fry-swift30'
    
    # Fabric
    
    pod 'Fabric'
    pod 'Crashlytics'
    
    # Deep linking
    pod 'Appz'
    pod 'Branch'
        
    # Bug repors
#    pod 'Instabug', '~> 7.2.4'
    
    # Hardware
    pod 'Device', '~> 3.0.1'
    
    # Google
    pod 'GooglePlaces'
    pod 'GoogleMaps'

**Comment by alexsanderkhitev** [comment]
@Kmohamed I also added  pod 'Chatto',  :git => 'https://github.com/badoo/Chatto.git', :branch => 'master'
 pod 'ChattoAdditions',  :git => 'https://github.com/badoo/Chatto.git', :branch => 'master'

**Comment by alexsanderkhitev** [comment]
@Kmohamed Please let us know when any details are known or the problem will be resolved.

**Comment by Kmohamed** [comment]
@alexsanderkhitev Sure.

**Comment by alexsanderkhitev** [comment]
Hello @Kmohamed! Is there any news on this issue?

**Comment by Kmohamed** [comment]
@alexsanderkhitev we are very sorry for being late.
It takes a lot of time because this issue is occurs randomly so we put the under a lot of testing cycles. But the good news it's almost done and will be released soon.

**Comment by alexsanderkhitev** [comment]
@Kmohamed thanks!

**Comment by Kmohamed** [comment]
@alexsanderkhitev
Sorry for being late in fixing this issue but the good news we just release a Fix for your issue please update to 7.3.5 and tell if you still face a problems.

**Comment by alexsanderkhitev** [comment]
@Kmohamed Thank you!
